src/core/songloader.cpp

```diff
@@ -674,21 +670,8 @@ bool SongLoader::LoadRemotePlaylist(const QUrl& url) {
     return false;
   }
 
-  // Save them to a temporary file...
-  QString playlist_filename =
-      Utilities::SaveToTemporaryFile(data_reply->readAll());
-  if (playlist_filename.isEmpty()) {
-    qLog(Error) << url.toString()
-                << "could not write contents to temporary file";
-    return false;
-  }
-
-  qLog(Debug) << url.toString() << "with MIME" << mime_type << "loading from"
-              << playlist_filename;
-
-  // ...and load it.
-  LoadPlaylist(parser, playlist_filename);
+  qLog(Debug) << "Loading" << url.toString() << "with MIME" << mime_type;
 
-  QFile(playlist_filename).remove();
+  songs_ = parser->Load(data_reply, QString(), QString());
```

Comment:
Minor: Load has default args for playlist_path and dir, so you shouldn't need the last two args. Also note that the last param is actually a QDir, not a QString.

Reply:
Done. I really do need to be more careful though...
Thanks~